core/include/core/serialization.h

```diff
@@ -65,14 +65,14 @@ struct g3frameobject_picklesuite : boost::python::pickle_suite
 };
 
 template <class T>
-inline int
+inline uint32_t
```

Comment:
Why uint32_t specifically, given the cast below, rather than a generic unsigned?

Reply:
¯\_(ツ)_/¯ why not?

Reply:
Because cereal [documents](https://uscilab.github.io/cereal/assets/doxygen/structcereal_1_1detail_1_1Version.html) it as being `uint32_t`? We're not likely to meet a compiler which implements `unsigned` as 16 bit, but I think it is allowed.